src/status_im/contexts/wallet/collectible/events.cljs

```diff
@@ -43,10 +43,16 @@
 
 (defn flush-collectibles
   [{:keys [db]}]
-  (let [collectibles-per-account (get-in db [:wallet :ui :collectibles :fetched])]
+  (let [collectibles-per-account (get-in db [:wallet :ui :collectibles :fetched])
+        updated-accounts         (move-collectibles-to-accounts (get-in db [:wallet :accounts])
+                                                                collectibles-per-account)
+        has-collectibles?        (some (fn [account]
+                                         (pos? (count (:collectibles account))))
+                                       (vals updated-accounts))]
     {:db (-> db
              (update-in [:wallet :ui :collectibles] dissoc :pending-requests :fetched)
-             (update-in [:wallet :accounts] move-collectibles-to-accounts collectibles-per-account))}))
+             (assoc-in [:wallet :accounts] updated-accounts)
+             (assoc :centralized-metrics/event-data has-collectibles?))}))
```

Comment:
It feels odd to me that we're `assoc`-ing additional data for metrics in the db here to pass extra data to the interceptor. Since we're already coupling the event itself with the metrics, i'd consider to imperatively send the metric with a separate event and not use the interceptor at all in such cases. I think it will be cleaner, less indirect and easier to understand what metrics are sent e.g.

```clj
{:db (-> ...)
 :fx [[:dispatch [:centralized-metrics/send-metric "flush-collectibles" {:has-collectibles? has-collectibles?}]]]}
```

I think the interceptor is good for cases that are more generic e.g. navigation, but given that we might have more complex requirements for metrics, which might need extra computation/data (like we have here) that is unavailable in the event args, maybe we should consider alternative approaches. Also, the way it's done here mixes the data required for the event and the data needed for metrics, which IMO should be kept apart.

Reply:
> It feels odd to me that we're assoc-ing additional data for metrics in the db here to pass extra data to the interceptor.

It is odd indeed, there's more than one way to solve the problem. And coming back to the problem, I initially started suggesting a different solution from the original in the PR because the `context/centralized_metrics/*` namespaces shouldn't know how to compute anything from other domains, therefore, the data should come already cooked and mostly ready to be sent to MixPanel.

Dispatching a separate event as you suggest can work well, I agree 👍🏼 What should the handler body look like? I'm imagining the body would use a new re-frame effect to execute the effectful call to `native-module/add-centralized-metric` (basically what we do in the global interceptor).

> less indirect and easier to understand what metrics

It's less magical, but it's more indirect technically because it will go through the whole re-frame loop and have more overhead (although marginal).

> Also, the way it's done here mixes the data required for the event and the data needed for metrics

I don't see how that's different than the alternative you suggested. The computed data is still present in the same event handler. One solution assocs to a completely separate root key, so it's not mixed with the event data. The alternative you provided sends the data to a separate event, but the computation still happens in the originating event. Both solutions couple in the correct place IMO.

I don't have any strong opinion tbh @clauxx, my only more strong opinion was that we shouldn't compute business data in the `centralized_metrics/` namespaces and your solution also solves that 👍🏼

Would be helpful for @vkjr if you could provide the full implementation suggestion so that we can get this PR merged soon. Maybe pair with him since you have the idea in your head?

Reply:
@ilmotta afaik @vkjr is off till the end of the week, but we discussed with @mohsen-ghafouri, and he will make a draft PR with a more specific implementation of what I explained, but for dapps metrics, and we can discuss it there.

Reply:
Sounds great, thanks @clauxx and @mohsen-ghafouri